spec.html

```diff
@@ -5577,17 +5577,12 @@ <h1>
       </h1>
       <dl class="header">
         <dt>description</dt>
-        <dd>It converts _value_ to a non-negative integer if the corresponding decimal representation, as a String, is an integer index.</dd>
+        <dd>It converts _value_ to an integer and returns that integer if it is non-negative and corresponds with an integer index. Otherwise, it throws an exception.</dd>
       </dl>
       <emu-alg>
-        1. If _value_ is *undefined*, then
-          1. Return 0.
-        1. Else,
-          1. Let _integer_ be ? ToIntegerOrInfinity(_value_).
-          1. Let _clamped_ be ! ToLength(𝔽(_integer_)).
-          1. If SameValue(𝔽(_integer_), _clamped_) is *false*, throw a *RangeError* exception.
-          1. Assert: 0 ≤ _integer_ ≤ 2<sup>53</sup> - 1.
-          1. Return _integer_.
+        1. Let _integer_ be ? ToIntegerOrInfinity(_value_).
+        1. If _integer_ is in the inclusive interval from 0 to 2<sup>53</sup> - 1, return _integer_.
```

Comment:
We seem to have an affinity for inclusive intervals, even when they make us offset a bound by 1. Personally, I would do this:

```suggestion
        1. If _integer_ is in the interval from 0 (inclusive) to 2<sup>53</sup> (exclusive), return _integer_.
```

But I think @syg prefers the way you have?

Reply:
Agreed, I prefer the existing text.